src/Environments/MacOS/ResourceFork.cs

```diff
@@ -342,7 +342,151 @@ public void Dump()
                 platform.A5World = a5world;
                 platform.A5Offset = jt.BelowA5Size;
                 segmentMap.AddSegment(a5world);
+
+                // Search segmentMaps for %A5Init CODE Segment and unpack global data to A5World
+                foreach ( var a5dataSegment  in segmentMap.Segments.Values)
+                {
+                    if (a5dataSegment.Name.Length >= 12 )
+                    {
+                        if (a5dataSegment.Name.Substring(5,7) == "%A5Init")
+                        {
```

Comment:
What about https://github.com/uxmal/reko/pull/591#discussion_r177093255?

Reply:
I am scanning the segments looking for the A5%Init code segment, not sure of a better way.
I did remove the embedded expansion DO loop into a seperate procedure.

Reply:
I just noticed in testing that I left a wrong value for bank size, it should be 0x00010000. So this needs to be updated. The testing was to check as I tried to display an output message if there was a problem with the information found in the A5%Init code segment. Not sure of the right way yet.

Reply:
Would not be better to write

```
if (a5dataSegment.Name.Length < 12 )
    continue;
if (a5dataSegment.Name.Substring(5,7) != "%A5Init")
   continue;
if (<somethig-else>)
  continue;
...
<A5Init found>
```
to avoid nested `if`s

Reply:
The other possible solution is split it to separate methods. Something like
```
var seg = findA5InitSegment(<args>);
if (seg != null)
    readA5InitSegment(seg, <other args>);
```